core/src/network/reducers/respond_get.rs

```diff
@@ -0,0 +1,61 @@
+use boolinator::*;
+use crate::{
+    action::ActionWrapper,
+    context::Context,
+    network::{actions::ActionResponse, state::NetworkState},
+};
+use holochain_core_types::{entry::Entry, error::HolochainError};
+use holochain_net_connection::{
+    net_connection::NetConnection,
+    protocol_wrapper::{DhtData, GetDhtData, ProtocolWrapper},
+};
+use std::sync::Arc;
+
+fn inner(
+    network_state: &mut NetworkState,
+    get_dht_data: &GetDhtData,
+    maybe_entry: &Option<Entry>,
+) -> Result<(), HolochainError> {
+    (network_state.network.is_some()
+        && network_state.dna_hash.is_some() & network_state.agent_id.is_some())
+    .ok_or("Network not initialized".to_string())?;
+
+    let data = DhtData {
+        msg_id: get_dht_data.msg_id.clone(),
+        dna_hash: network_state.dna_hash.clone().unwrap(),
+        agent_id: get_dht_data.from_agent_id.clone(),
+        address: get_dht_data.address.clone(),
+        content: serde_json::from_str(&serde_json::to_string(&maybe_entry).unwrap()).unwrap(),
```

Comment:
isn't serde_json deprecated? Shouldn't you use JsonString instead?
Also, I'm not sure I understand the usefulness of serializing an option that could be None.

Reply:
The `DhtData` struct is defined with serde, `content` is of type `serde_json::Value`. Yeah, @thedavidmeister brought this up before. This is the network-module API as @neonphog built it for us. We should change this to JsonString but I didn't want to put that into the scope of this PR. It is still just a few lines like this.

The option is for being able to respond with "I don't have the requested entry".